service/telemetry.go

```diff
@@ -69,6 +72,18 @@ func telemetryFlags(flags *flag.FlagSet) {
 		metricsPrefixCfg,
 		"otelcol",
 		"Prefix to the metrics generated by the collector.")
+
+	useLegacyMetricsPtr = flags.Bool(
+		"legacy-metrics",
+		true,
+		"Flag to control usage of legacy metrics",
+	)
+
+	useNewMetricsPtr = flags.Bool(
+		"new-metrics",
+		false,
+		"Flag to control usage of new metrics",
+	)
```

Comment:
Can we use one command line option instead of 2? Shouldn't these flags be exclusive or we have a usecase where both are enabled or both are disabled?

Reply:
It is convenient for tests and during the transition to having both, the goal is to remove both before first official release.

/cc @rghetia